src/handler/active_requests.rs

```diff
@@ -38,7 +38,7 @@ impl ActiveRequests {
         match self.active_requests_nonce_mapping.remove(nonce) {
             Some(node_address) => match self.active_requests_mapping.remove(&node_address) {
                 Some(request_call) => Some((node_address, request_call)),
-                None => None,
```

Comment:
It's better to add a message here for two reasons
1. It serves as a comment explaining why this should not happen
2. In the eventuality that it happens, the panic is going to print that message